exercises/say/example.js

```diff
@@ -38,42 +38,47 @@ const bigNumbers = {
   1000000000: 'billion',
 };
 
-function bigPart(number) {
-  let factor,
-    result = '';
-  for (let bigNumber = 1000000000; bigNumber >= 1000; bigNumber /= 1000) {
-    if (number.current >= bigNumber) {
-      factor = Math.floor(number.current / bigNumber);
-      result += `${threeDigit(factor)} ${bigNumbers[bigNumber]} `;
-      number.current -= factor * bigNumber;
-    }
-  }
-  return result;
-}
-
 const sayDecade = (n) => {
   for (let decade = 90; decade >= 20; decade -= 10) {
     if (n >= decade) {
       return `${decades[decade]}-${smallNumbers[n - decade]}`;
     }
   }
+  return '';
 };
 
-const twoDigit = n => n < 20 ? smallNumbers[n] : sayDecade(n);
-
-const threeDigit = n => n < 100 ? twoDigit(n)
-  : `${smallNumbers[Math.floor(n / 100)]} hundred ${twoDigit(n % 100)}`;
+function twoDigit(n) {
+  return n < 20 ? smallNumbers[n] : sayDecade(n);
+}
 
-export default class Say {
-  inEnglish(n) {
-    let result;
-    const number = { current: n };
+function threeDigit(n) {
+  return n < 100 ? twoDigit(n)
+    : `${smallNumbers[Math.floor(n / 100)]} hundred ${twoDigit(n % 100)}`;
+}
 
-    if (n >= 0 && n < 1000000000000) {
-      result = bigPart(number);
-      result += threeDigit(number.current);
-      return result.replace(/.zero/, '');
+function bigPart(number) {
+  let factor;
+  let result = '';
+  const num = number;
+  for (let bigNumber = 1000000000; bigNumber >= 1000; bigNumber /= 1000) {
+    if (num.current >= bigNumber) {
+      factor = Math.floor(num.current / bigNumber);
+      result += `${threeDigit(factor)} ${bigNumbers[bigNumber]} `;
+      num.current -= factor * bigNumber;
     }
-    throw new Error('Number must be between 0 and 999,999,999,999.');
   }
+  return result;
 }
+export default class Say {
+}
+Say.prototype.inEnglish = (n) => {
```

Comment:
Since you are using prototype, `class` is not really required. A function should be sufficient. eg. https://github.com/exercism/javascript/blob/5062905a6b635623cc1914c519083925e661ea1e/exercises/all-your-base/example.js#L27-L29

Reply:
Should we instead use class, since this is ES6 now?  Wondering if anyone has an opinion on this: `Function.prototype` feels more ES5 and prior to me, although this PR is for linting, so I don't want to conflate the two issues unless it's sensible to address them at the same time.

Reply:
> although this PR is for linting, so I don't want to conflate the two issues unless it's sensible to address them at the same time.

Exactly my thoughts. I saw prototype being used on few other exercises as well. Let us create separate issue to remove prototypes from all of them.

Reply:
+1

Reply:
@tejasbubane @matthewmorgan I made the changes to use ES6 classes.
- sorry for the delay 👼

Reply:
@msomji thanks for your updates!